src/ethereum/crypto/elliptic_curve.py

```diff
@@ -0,0 +1,117 @@
+"""
+Elliptic Curves
+^^^^^^^^^^^^^^^
+"""
+
+from typing import Generic, Type, TypeVar
+
+from .finite_field import Field
+
+F = TypeVar("F", bound=Field)
+T = TypeVar("T", bound="EllipticCurve")
+
+
+class EllipticCurve(Generic[F]):
+    """
+    Superclass for integers modulo a prime. Not intended to be used
+    directly, but rather to be subclassed.
+    """
+
+    __slots__ = ("x", "y")
+
+    FIELD: Type[F]
+    A: F
+    B: F
+
+    x: F
+    y: F
+
+    def __new__(cls: Type[T], x: F, y: F) -> T:
+        """
+        Make new point on the curve. The point is not checked to see if it is
+        on the curve.
+        """
+        res = object.__new__(cls)
+        res.x = x
+        res.y = y
+        return res
+
+    def __init__(self, x: F, y: F) -> None:
+        """
+        Checks if the point is on the curve. To skip this check call
+        `__new__()` directly.
+        """
+        if (
+            x != self.FIELD.zero() or y != self.FIELD.zero()
+        ) and y ** 2 - x ** 3 - self.A * x - self.B != self.FIELD.zero():
+            raise ValueError("Point not on curve")
+
+    def __eq__(self, other: object) -> bool:
+        """
+        Test two points for equality.
+        """
+        if not isinstance(other, type(self)):
+            return False
+        return self.x == other.x and self.y == other.y
+
+    def __str__(self) -> str:
+        """
+        Stringify a point as its coordinates.
+        """
+        return str((self.x, self.y))
+
+    @classmethod
+    def inf(cls: Type[T]) -> T:
```

Comment:
`infinity` perhaps, unless there's a good reason not to spell it out.

Reply:
`point_at_infinity()`